src/analysis/mesh/qgsmeshtriangulation.cpp

```diff
@@ -0,0 +1,339 @@
+/***************************************************************************
+                          qgsmeshtriangulation.cpp
+                          -----------------
+    begin                : August 9th, 2020
+    copyright            : (C) 2020 by Vincent Cloarec
+    email                : vcloarec at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsmeshtriangulation.h"
+#include "qgsdualedgetriangulation.h"
+#include "qgsvectorlayer.h"
+#include "qgscoordinatetransformcontext.h"
+#include "qgscurve.h"
+#include "qgscurvepolygon.h"
+#include "qgsmultisurface.h"
+#include "qgsmulticurve.h"
+#include "qgsfeedback.h"
+#include "qgslogger.h"
+
+QgsMeshTriangulation::QgsMeshTriangulation(): QObject()
+{
+  mTriangulation.reset( new QgsDualEdgeTriangulation() );
+}
+
+
+QgsMeshTriangulation::~QgsMeshTriangulation() = default;
+
+bool QgsMeshTriangulation::addVertices( QgsFeatureIterator &vertexFeatureIterator, int valueAttribute, const QgsCoordinateTransform &transform, QgsFeedback *feedback, int featureCount )
+{
+  if ( !vertexFeatureIterator.isValid() || vertexFeatureIterator.isClosed() )
+    return false;
+
+  if ( feedback )
+    feedback->setProgress( 0 );
+
+  QgsFeature feat;
+  int i = 0;
+  while ( vertexFeatureIterator.nextFeature( feat ) )
+  {
+    if ( feedback )
+    {
+      if ( feedback->isCanceled() )
+        break;
+
+      feedback->setProgress( 100 * i / featureCount );
+      i++;
+    }
+
+    addVerticesFromFeature( feat, valueAttribute, transform, feedback );
+  }
+
+  return true;
+}
+
+bool QgsMeshTriangulation::addBreakLines( QgsFeatureIterator &lineFeatureIterator, int valueAttribute, const QgsCoordinateTransform &transform, QgsFeedback *feedback, int featureCount )
+{
+  if ( !lineFeatureIterator.isValid() || lineFeatureIterator.isClosed() )
+    return false;
+
+  if ( feedback )
```

Comment:
Some places here you check if feedback is nullptr, elsewhere you don't -- if it's optional better guard all uses (also in the other methods in this file)

Reply:
Maybe I missed something, but I don't see any place where there is no check.

Reply:
Sorry, my fault -- read the diff badly! You're all clear here